### PR TITLE
[FIX] Channels selection in leave team modal not working

### PR DESCRIPTION
**Pull request by Darshilp326**
## Proposed changes (including videos or screenshots)

Users would be able to select team channels that they want to leave while leaving the team.
Users would now automatically leave from team channels he/she selected.

BEFORE

https://user-images.githubusercontent.com/55157259/116005794-24205980-a626-11eb-888e-59e42ca21f9a.mp4

BEFORE

https://user-images.githubusercontent.com/55157259/115970150-4babef00-a55e-11eb-97c9-45683ac16cc1.mp4

AFTER

https://user-images.githubusercontent.com/55157259/116005344-21bd0000-a624-11eb-8e58-a956579a08b4.mp4

AFTER

https://user-images.githubusercontent.com/55157259/115970155-5070a300-a55e-11eb-877f-862dd9a974d7.mp4

### CODE REVIEW

**Comment by Darshilp326** [comment]
@dougfabris  @ggazzo Please review :smile:

**Comment by dougfabris** [comment]
@Darshilp326 Amazing fix thank you! We could improve more and just send the user to the second modal's step if he can't select any channel. If he can, we need to show the channel's selection step first. Can you help with that?

**Comment by Darshilp326** [comment]
> @Darshilp326 Amazing fix, thank you! We could improve more and just send the user to the second modal's step if he can't select any channel. If he can, we need to show the channel's selection step first. Can you help with that?

Yes, sure!

File: app/api/server/v1/teams.ts [hunk 1]
```diff
@@ -172,7 +172,7 @@ API.v1.addRoute('teams.listRoomsOfUser', { authRequired: true }, {
 
 		const allowPrivateTeam = hasPermission(this.userId, 'view-all-teams', team.roomId);
 
-		if (!hasPermission(this.userId, 'view-all-team-channels', team.roomId)) {
+		if (!hasPermission(this.userId, 'view-all-team-channels', team.roomId) && userId !== this.userId) {
```
Comment (KevLehman) [inline app/api/server/v1/teams.ts:175]: Hey! Talking with Diego, the change looks good 

Would you mind adding some e2e tests to this endpoint to cover this behavior? 🙏

**Review by KevLehman — CHANGES_REQUESTED** [review]

File: app/api/server/v1/teams.ts (hunk 1 above)
Comment (dougfabris) [reply app/api/server/v1/teams.ts:175]: @Darshilp326 Are you planning to continue work on this one?

**Comment by dougfabris** [comment]
We ended up in a refactor for this LeaveTeam function and this fix is on there also #27197
Thanks a lot for your contribution @Darshilp326

**Pull request closed without merging**